sources/tools.c

```diff
@@ -0,0 +1,48 @@
+/**
+ * @file tools.c
+ * 
+ * @author Sam Donnelly (samueldonnelly11@gmail.com)
+ * 
+ * @brief General code tools used across files 
+ * 
+ * @version 0.1
+ * @date 2022-09-14
+ * 
+ * @copyright Copyright (c) 2022
+ * 
+ */
+
+//=======================================================================================
+// Includes 
+
+#include "tools.h"
+
+//=======================================================================================
+
+
+//=======================================================================================
+// String functions 
+
+// String comparison 
+uint8_t str_compare(
+    char *ref_msg, 
+    char *msg, 
+    uint8_t msg_start)
+{
+    // Local variables 
+    uint8_t str_length = strlen(ref_msg); 
```

Comment:
This can cause problem if 'ref_msg' doesn't have a NULL termination, like if it were garbage data for example.

Consider using 'strnlen()' as a safeguard against this